lib/vanagon/driver.rb

```diff
@@ -76,15 +76,16 @@ def run
       @engine.startup(@workdir)
 
       puts "Target is #{@engine.target}"
-
-      install_build_dependencies
-      @project.fetch_sources(@workdir)
-      @project.make_makefile(@workdir)
-      @project.make_bill_of_materials(@workdir)
-      @project.generate_packaging_artifacts(@workdir)
-      @engine.ship_workdir(@workdir)
-      @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
-      @engine.retrieve_built_artifact
+      Vanagon::Utilities.retry_with_timeout(3, 3600) do
+        install_build_dependencies
+        @project.fetch_sources(@workdir)
+        @project.make_makefile(@workdir)
+        @project.make_bill_of_materials(@workdir)
+        @project.generate_packaging_artifacts(@workdir)
+        @engine.ship_workdir(@workdir)
+        @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
+        @engine.retrieve_built_artifact
+      end
```

Comment:
I was just writing a note to that effect, wanted to talk over with the team. Wrap each item? break them down into smaller chinks?